server/src/schema/users.ts

```diff
@@ -42,11 +44,40 @@ export const SignInUserSchema = UserSchema.pick({
   password: true,
 });
 
-export const UpdateUserSchema = UserSchema.pick({
-  name: true,
-  password: true,
+export const UpdateUserSchema = z.object({
+  name: z
+    .string()
+    .min(3, { message: "Name must be at least 3 characters" })
+    .optional(),
+  email: z
+    .string()
+    .email("Must be a valid email")
+    .includes("srmist.edu.in", {
+      message: "Email must be a SRMIST email",
+    })
+    .optional(),
+  registrationNumber: z
+    .string()
+    .includes("RA", {
+      message: "Registration number must include 'RA'",
+    })
+    .length(15, { message: "Registration number must be 15 characters" })
+    .optional(),
+  password: z
+    .string()
+    .min(8, {
+      message: "Password must be at least 8 characters",
+    })
+    .optional(),
+}).refine((data) => Object.keys(data).length > 0, {
+  message: "At least one field must be provided for update",
```

Comment:
why this change? the user must not be able to change anything except name and password

Reply:
![image](https://github.com/user-attachments/assets/82531019-77d9-4cfe-af8a-5fdaf7a8c16a)